### PR TITLE
Enable WebPush on Mac

### PR DESCRIPTION
**Pull request by aosmond**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -215,10 +215,12 @@ You'll need some dependencies installed to build.
 $ brew install openssl libupnp sqlite
 ```
 
-This is required to build the openssl crate using homebrew's openssl:
+This is required to build/link the openssl crate and foxbox using homebrew's openssl:
 
 ``` bash
 $ export DEP_OPENSSL_INCLUDE=/usr/local/opt/openssl/include/
+$ export OPENSSL_LIB_DIR=/usr/local/opt/openssl/lib
+$ export EXTRA_LDFLAGS=-L/usr/local/opt/openssl/lib
```
Comment (ferjm) [inline README.md:223]: Do we need to add these exports to the .travis.yml file?

**Comment by Yoric** [comment]
@aosmond I'm trying to understand: why should this fix the issue?

**Comment by Yoric** [comment]
As far as I can tell, this is not going to change anything on my computer. I have built rust-openssl manually (using their `cargo build`) and it already uses my version of openssl that lives in `/usr/local/opt/openssl`.

If you wish to test which version is used by Foxbox, I have a patch pending review in rust-openssl: https://github.com/sfackler/rust-openssl/pull/380#issuecomment-209640698

**Comment by aosmond** [comment]
@Yoric That doesn't seem to be the case for others since they have build issues. Not being a Mac user I find speculating difficult. Feel free to update the README :).

**Comment by nojunpark** [comment]
I think when brew is used to install openssl (like the readme suggests) , such export is necessary, but I guess in case of macports it might be different. Then again I had a few issues with installing openssl using macports so I went back to homebrew.

**Comment by Yoric** [comment]
OK. Unfortunately, it won't fix #358, though.

**Comment by nojunpark** [comment]
I think #358 doesn't happen with brew install method - I remember seeing occasional unwrap() before, but now it's not happening.

**Comment by Yoric** [comment]
Well, we found out that #358 is due to using the wrong version of curl (and not handling failures too gracefully).

**Pull request merged**